### PR TITLE
Added git, web and docker download sources

### PR DESCRIPTION
**Pull request by lewismarshall**

### CODE REVIEW

File: pkg/cmd/artefactor.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package cmd
```
Comment (KashifSaadat) [inline pkg/cmd/artefactor.go]: Rename repository to `artefactor` (or all code references to `artifactor`), for consistency.

**Comment by jon-shanks** [comment]
Can we create some docs for this as well, even though a WIP just to give a basic overview of it and how to use it, can improve them over time.

File: pkg/cmd/artefactor.go (hunk 1 above)
Comment (KashifSaadat) [reply pkg/cmd/artefactor.go]: README can also do with an update including description, example use cases and how-to etc.

Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: All done - it's UK though so artefactor.

File: LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+The MIT License (MIT)
+=====================
+
+Copyright (c) 2015
```
[the remark this answers is not in the document]
Comment (lewismarshall) [reply LICENSE.md]: Nice one - pushed update

File: pkg/cmd/artefactor.go [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
 const (
+	EnvPrefix = "ARTEFACTOR_"
 	// FlagDockerImages specifies a to newline seperated list of docker images to save
 	FlagDockerImages = "docker-images"
-	// The directory to save archives into
+	// FlagArchiveDir is the directory to save archives into
 	FlagArchiveDir = "archive-dir"
+	// FlagGitRepos specifies a newline speerated list of local or remote git repos
+	FlagGitRepos = "git-repos"
+	// FlagWebFiles specifies a whitespace delimited set of csv's with url,file,md5
```
Comment (KashifSaadat) [inline pkg/cmd/artefactor.go]: With your example in the README there's also a bool on the end?

File: pkg/cmd/publish.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		publishCmd,
+		FlagArchiveDir,
+		DefaultArchiveDir,
+		"a directory where artefacts exist to publish from")
+
+	addFlagWithEnvDefault(
+		publishCmd,
+		FlagDockerRegistry,
+		"",
+		"where to publish images e.g. myrestirty.com/path")
```
Comment (KashifSaadat) [inline pkg/cmd/publish.go]: was this meant to say myregistry? :)

File: pkg/cmd/restore.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+	}
+
+	// Get the list of files from the checksum file (hashcache) in the source
+	// directory.
+	dstDir := filepath.Join(dst, repoName, savedDir)
+	files := hashcache.GetFiles(src)
+	// Create directory structure
+	if _, err := os.Stat(dstDir); err != nil {
+		if err := os.MkdirAll(dstDir, 0755); err != nil {
+			return err
```
Comment (KashifSaadat) [inline pkg/cmd/restore.go]: Return formatted error here with some context

File: pkg/cmd/save.go [hunk 6]
```diff
@@ -1,43 +1,127 @@
...
 	RootCmd.AddCommand(saveCmd)
 }
 
-func save(c *cobra.Command) {
+func save(c *cobra.Command) error {
+	common(c)
+	// Record where the archives should be stored
+	saveDir := c.Flag(FlagArchiveDir).Value.String()
+	if err := saveSavedPath(saveDir); err != nil {
+		return err
```
Comment (KashifSaadat) [inline pkg/cmd/save.go]: format error with context

File: pkg/cmd/save.go [hunk 7]
```diff
@@ -1,43 +1,127 @@
...
+		if err := docker.Save(image, saveDir); err != nil {
+			return err
+		}
+	}
+
+	// Next save any git repos
+	gitRepos := strings.Fields(c.Flag(FlagGitRepos).Value.String())
+	for _, repo := range gitRepos {
+		if err := git.Archive(repo, saveDir); err != nil {
+			return err
```
Comment (KashifSaadat) [inline pkg/cmd/save.go]: format error with context

File: pkg/cmd/artefactor.go (hunk 3 above)
Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: I've improved the readme and made the comment accurate.

File: pkg/cmd/publish.go (hunk 4 above)
Comment (lewismarshall) [reply pkg/cmd/publish.go]: Yep - updated to be correct and more descriptive.

File: pkg/cmd/restore.go (hunk 5 above)
Comment (lewismarshall) [reply pkg/cmd/restore.go]: Indeed, nice to be nice - improved all top level error messages.

File: pkg/cmd/save.go (hunk 6 above)
Comment (lewismarshall) [reply pkg/cmd/save.go]: Indeed, nice to be nice - improved all top level error messages.

File: pkg/cmd/save.go (hunk 7 above)
Comment (lewismarshall) [reply pkg/cmd/save.go]: Indeed, nice to be nice - improved all top level error messages.

**Pull request merged**